EndianTester/MainWindow.xaml.cs

```diff
@@ -0,0 +1,376 @@
+﻿using System;
+using System.IO;
+using System.Windows;
+using System.Diagnostics;
+using EndianTester.Utilities;
+using System.Buffers.Binary;
+
+namespace EndianTester
+{
+    /// <summary>
+    /// Interaction logic for MainWindow.xaml
+    /// </summary>
+    public partial class MainWindow : Window
+    {
+
+        /*    Results on XPS laptop i7-10875H  VS 2021 netcore 3.1
+         *    note binary primitives are slow, but easy to use....
+
+            >>>Testing Big Endian INT16 extraction times, 5242880 loops
+            Array.Reverse 16 ET = 129.715 mS, nS/loop = 24.741
+            BinaryPrimitives 16 ET = 134.741 mS, nS/loop = 25.700
+            Shift-OR 16 ET = 48.894 mS, nS/loop = 9.326
+
+            >>>Testing Big Endian INT32 extraction times, 2621440 loops
+            Array.Reverse 32 ET = 69.100 mS, nS/loop = 26.360
+            BinaryPrimitives 32 ET = 70.928 mS, nS/loop = 27.057
+            Shift-OR 32 ET = 25.128 mS, nS/loop = 9.586
+
+            >>>Testing Big Endian INT64 extraction times, 1310720 loops
+            Array.Reverse 64 ET = 38.235 mS, nS/loop = 29.171
+            BinaryPrimitives 32 ET = 36.075 mS, nS/loop = 27.523
+            Shift-OR 64 ET = 13.290 mS, nS/loop = 10.140
+
+            >>>Testing Big Endian FLOAT-32 extraction times, 2621440 loops
+            Array.Reverse FLOAT-32 ET = 77.382 mS, nS/loop = 29.519
+            Shift-OR FLOAT-32 ET = 42.955 mS, nS/loop = 16.386
+
+            >>>Testing Big Endian FLOAT-64 extraction times, 1310720 loops
+            Array.Reverse FLOAT-64 ET = 41.860 mS, nS/loop = 31.937
+            Shift-OR FLOAT-64 ET = 21.677 mS, nS/loop = 16.538
+        */
```

Comment:
Awesome that you went out and measured which way is faster. Profiling dynamic runtimes is a remarkably difficult task. There are a ton of pitfalls. Although a Java centric article, I'm going to link it here as many of the points apply to Roslyn in the .NET world as well [1].

For example the `val` local variable inside of your `for` loops is not used anywhere, so a semi-smart JIT compiler will actually fold away the entire `for` loop as part of dead code elimination optimization pass. I wouldn't be surprised if something like this is actually happening.

In any way while a useful exercise to profile I don't think we need to contribute this code to the project as it doesn't provide much value to the library.

Do you think you could limit this PR to only the changes which modify how bytes are ready from the `ByteReader`?

[1] https://www.oracle.com/technical-resources/articles/java/architect-benchmarking.html